lib/gs-plugin-loader.c

```diff
@@ -2505,10 +2505,11 @@ mount_removed_cb (GVolumeMonitor *volume_monitor,
 		  GsPluginLoader *plugin_loader)
 {
 	GsPluginLoaderPrivate *priv = gs_plugin_loader_get_instance_private (plugin_loader);
-	GList *list_orig = priv->removable_mounts;
+	guint orig_len = g_list_length (priv->removable_mounts);
 
-	priv->removable_mounts = g_list_remove (priv->removable_mounts, mount);
-	if (list_orig != priv->removable_mounts)
+	if (priv->removable_mounts != NULL)
+		priv->removable_mounts = g_list_remove (priv->removable_mounts, mount);
+	if (orig_len != g_list_length (priv->removable_mounts))
```

Comment:
This Is why I think it'd be better to keep a `has_mount` function as before, or a `remove_if_exists -> boolean`.